include/picongpu/particles/atomicPhysics/kernel/CalculateRejectionProbability.hpp

```diff
@@ -0,0 +1,122 @@
+/* Copyright 2024 Brian Marre
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/defines.hpp"
+
+#include <cstdint>
+
+namespace picongpu::particles::atomicPhysics::kernel
+{
+    /** methods for calculating the rejection probability
+     *
+     * @return sets the rejection probability cache entry for the specified entry to either
+     *  -1, no rejection probability necessary
+     *  >= 0, rejection probability
+     *
+     * also sets the flag sharedResourcesOverSubscribed to true if an over subscription is found
+     */
+    struct CalculateRejectionProbability
+    {
+        using VectorIdx = DataSpace<picongpu::simDim>;
+
+        template<typename T_Histogram, typename T_RejectionProbabilityCache>
+        HDINLINE static void ofHistogramBin(
+            uint32_t const binIndex,
+            T_Histogram const& histogram,
+            T_RejectionProbabilityCache& rejectionProbabilityCache,
+            bool& sharedResourcesOverSubscribed)
+        {
+            float_X const weight0 = histogram.getBinWeight0(binIndex);
```

Comment:
please set `sharedResourcesOverSubscribed=false;` because in the beginning because you can not assume that the value is correctly set by the caller. If you return this value the caller can not call the method wrong at all.

Reply:
done